tools/partition_graph_lp.py

```diff
@@ -149,15 +156,43 @@
         target_etypes = [target_etypes]
 
     if constructed_graph:
+        d_shuffled_nids = {} # to store shuffled nids by ntype to avoid different orders for the same ntype
         for target_e in target_etypes:
             num_edges = g.num_edges(target_e)
             g.edges[target_e].data['train_mask'] = th.full((num_edges,), False, dtype=th.bool)
             g.edges[target_e].data['val_mask'] = th.full((num_edges,), False, dtype=th.bool)
             g.edges[target_e].data['test_mask'] = th.full((num_edges,), False, dtype=th.bool)
-            g.edges[target_e].data['train_mask'][: int(num_edges * args.train_pct)] = True
-            g.edges[target_e].data['val_mask'][int(num_edges * args.train_pct): \
-                                               int(num_edges * (args.train_pct + args.val_pct))] = True
-            g.edges[target_e].data['test_mask'][int(num_edges * (args.train_pct + args.val_pct)): ] = True
+            if not args.inductive_split:
+                # Randomly split links
+                g.edges[target_e].data['train_mask'][: int(num_edges * args.train_pct)] = True
+                g.edges[target_e].data['val_mask'][int(num_edges * args.train_pct): \
+                                                int(num_edges * (args.train_pct + args.val_pct))] = True
+                g.edges[target_e].data['test_mask'][int(num_edges * (args.train_pct + args.val_pct)): ] = True
+            else:
+                # Inductive split for link prediction
```

Comment:
Does this mean some head nodes are not available to the training graph?
Can you add some details here?

Reply:
> Does this mean some head nodes are not available to the training graph?
Yes. 
I'll add more comments.